src/main/frontend/modules/outliner/transaction.cljc

```diff
@@ -0,0 +1,28 @@
+(ns frontend.modules.outliner.transaction
+  #?(:cljs (:require-macros [frontend.modules.outliner.transaction])))
+
+(defmacro transact!
+  "Batch all the transactions in `body` to a single transaction, Support nested transact! calls.
+  Currently there are no options, it'll execute body and collect all transaction data generated by body, then calls `listeners` on the `tx-report`.
+  (transact! {:repo \"test\"}
```

Comment:
what :repo option for? and how many options this macro support? need more docs here

Reply:
Updated [here](https://github.com/logseq/logseq/pull/4880/files#diff-e4efca409ffb802ba3fe8871feb01826355c192103ad7a6b34574697d97182b9R7-R12).